### PR TITLE
Int 578 fix dob search with dashes

### PR DESCRIPTION
**Pull request by anuroop15**
### Jira Story

- [dashes in DOB/SSN not providing users with search results INT-578](https://osi-cwds.atlassian.net/browse/INT-578)

### Purpose
updates person search query to remove dases ('-') in DOB and add replace with space in name having dashes.

### CODE REVIEW

File: app/services/person_search_query_builder.rb [hunk 1]
```diff
@@ -6,7 +6,10 @@ class PersonSearchQueryBuilder
   attr_reader :search_term
 
   def initialize(search_term = '')
-    @search_term = search_term.downcase.delete('/')
+    new_search = search_term.split(' ').map do |n|
+      n.to_i != 0 ? n.gsub(/[^0-9A-Za-z]/, '') : n.gsub(/[^0-9A-Za-z]/, ' ')
+    end
```
Comment (AmanShurafa) [inline app/services/person_search_query_builder.rb]: The code is not clear:
   - `n` is not a descriptive variable name.
   - I'm not sure `.to_i != 0` is enough to detect if search term is ssn or dob. I would feel better if we did a regex match `/(\d+\W?){3}/.match(search_term) ? 'dob or ssn' : 'name'`.

I suggest something like this:
```
dob_ssn_regex = /(\d+\W?){3}/
terms_seperator = dob_ssn_regex.match(search_term) ? '' : ' '
@search_term = search_term.gsub(/[\W_]/, terms_seperator).downcase
```

File: app/services/person_search_query_builder.rb [hunk 2]
```diff
@@ -6,7 +6,10 @@ class PersonSearchQueryBuilder
...
+    @search_term = new_search.join(' ').downcase.delete('/')
```
Comment (AmanShurafa) [inline app/services/person_search_query_builder.rb]: No need for `.delete('/'`. It's already taken care of.

File: spec/services/people_search_query_builder_spec.rb [hunk 3]
```diff
@@ -3,44 +3,52 @@
 require 'rails_helper'
 
 describe PersonSearchQueryBuilder do
```
Comment (AmanShurafa) [inline spec/services/people_search_query_builder_spec.rb]: I like how clean you made the test but can you add more test cases for various DoB and SSN formats to ensure the code does what we think it does.

File: spec/support/helpers/autocompleter_helpers.rb [hunk 4]
```diff
@@ -26,7 +26,7 @@ def populate_autocompleter_with_options(locator, value, split)
     if split
       value.split('').each do |character|
         find_field(locator).native.send_keys(character)
-        sleep 0.41
+        sleep 0.3
```
Comment (AmanShurafa) [inline spec/support/helpers/autocompleter_helpers.rb]: What's the reason for changing this?

File: spec/features/screening/participant/search_participant_spec.rb [hunk 5]
```diff
@@ -4,8 +4,111 @@
 require 'spec_helper'
 require 'feature/testing'
 feature 'searching a participant in autocompleter' do
+  def build_response_from_person(person)
```
Comment (AmanShurafa) [inline spec/features/screening/participant/search_participant_spec.rb]: 👍 I like this, would prefer to move it to `spec_helper` so others can find it and use it as well. Not necessary though.

File: spec/features/screening/participant/search_participant_spec.rb [hunk 6]
```diff
@@ -70,36 +137,87 @@
...
+      expect(
+        a_request(
+          :post,
+          dora_api_url(Rails.application.routes.url_helpers.dora_people_light_index_path)
+        )
+            .with('body' => {
+                    'query' => {
+                      'bool' => {
+                        'must' => array_including(
+                          'multi_match' => hash_including('query' => 'ma 12323')
```
Comment (AmanShurafa) [inline spec/features/screening/participant/search_participant_spec.rb]: 👍 Took me a while to find this line. It would help if you do the refactor in a separate commit to make it easier to find the changes relevant to the story you're working on.

File: spec/features/screening/participant/search_participant_spec.rb [hunk 7]
```diff
@@ -70,36 +137,87 @@
         expect(page).to have_content 'Home'
         expect(page).to have_content '971-287-6774'
         expect(page).to have_content 'SSN'
+        expect(page).to have_content '1234'
```
Comment (AmanShurafa) [inline spec/features/screening/participant/search_participant_spec.rb]: 👍

File: spec/support/helpers/autocompleter_helpers.rb (hunk 4 above)
Comment (anuroop15) [reply spec/support/helpers/autocompleter_helpers.rb]: as we are writing feature specs on "ma 123-23",split method is called with sleep time of 0.41. This is making stub call to time out. so I have decreased the sleep time to 0.3

File: spec/features/screening/participant/search_participant_spec.rb (hunk 5 above)
Comment (anuroop15) [reply spec/features/screening/participant/search_participant_spec.rb]: it seems like there was a change in build response. was the extraction needed?
https://github.com/ca-cwds/intake/blob/master/spec/features/screening/participant/search_participant_spec.rb#L81

File: spec/features/screening/participant/search_participant_spec.rb (hunk 6 above)
Comment (anuroop15) [reply spec/features/screening/participant/search_participant_spec.rb]: I think I made feature spec as a separate commit. I have changed the sleep time to 0.3 which was related to following feature spec. Does it need to be separated out from autocompleter helper??

File: app/services/person_search_query_builder.rb (hunk 2 above)
Comment (anuroop15) [reply app/services/person_search_query_builder.rb]: 👍

File: spec/services/people_search_query_builder_spec.rb (hunk 3 above)
Comment (anuroop15) [reply spec/services/people_search_query_builder_spec.rb]: added 👍

File: app/services/person_search_query_builder.rb (hunk 1 above)
Comment (anuroop15) [reply app/services/person_search_query_builder.rb]: - will change the the variable name.
- As discussed, the match regex doesn't join out search term. So doesn't work for out requirement.
